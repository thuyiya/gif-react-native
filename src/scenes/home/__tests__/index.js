/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import HomeScreen from "_scenes/home";
import { VIEW_CONST } from "_constants";
import { Giphy } from "_models";

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  route: {
    params: {
      item: new Giphy({}),
    },
  },
  ...props,
});

describe("SCREEN: Home", () => {
  const props = createTestProps({});
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<HomeScreen {...props} />);
  });

  it("enders correctly react-test-renderer", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Child Element Getting Rendered Correctly", () => {
    expect(
      wrapper.find({ testID: VIEW_CONST.DETAILS.giftCardTestID }).length
    ).toBe(1);
  });
});
