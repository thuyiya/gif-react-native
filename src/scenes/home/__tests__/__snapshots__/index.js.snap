// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCREEN: Home enders correctly react-test-renderer 1`] = `ShallowWrapper {}`;
